Jenkinsfile

```diff
@@ -82,102 +25,66 @@ pipeline {
-                                break
-                        }
-
-                        if (Globals.build || Globals.deploy || Globals.runTrivyScan) {
-                            echo 'Starting with calculating version'
-                            def shortBranchName = env.BRANCH_NAME.replaceAll("[^a-zA-Z0-9]+", "").take(30).toLowerCase()
-                            try {
-                                Globals.version = sh(script: "git describe --tags --match v[0-9]*", returnStdout: true).trim()
-                            } catch (err) {
-                                def version = sh(script: "git rev-parse --short HEAD", returnStdout: true).trim()
-                                Globals.version = "${shortBranchName}-${version}"
-                            }
-                            echo "Using version ${Globals.version}"
-                            if (env.BRANCH_NAME == 'main') {
-                                Globals.imageTag = "${Globals.IMAGE_NAME}:latest"
-                            } else {
-                                Globals.imageTag = "${Globals.IMAGE_NAME}:${shortBranchName}"
-                            }
-                            Globals.awsEcrRepo = "${env.AWS_ACCOUNT_ID}.dkr.ecr.${Globals.AWS_REGION}.amazonaws.com"
-                            Globals.awsEcrImageTag = "${Globals.awsEcrRepo}/${Globals.awsImageName}:${shortBranchName}"
-                            echo "Using container version ${Globals.imageTag}"
-                            echo "Using awsEcrRepo ${Globals.awsEcrRepo}"
+                script {
+                    echo '---- INSTALL MCHBUILD ----'
+                    sh '''
+                    python -m venv .venv-mchbuild
+                    PIP_INDEX_URL=https://hub.meteoswiss.ch/nexus/repository/python-all/simple \
+                      .venv-mchbuild/bin/pip install --upgrade mchbuild
+                    '''
+                    echo '---- INITIALIZE PARAMETERS ----'
+
+                    if (env.TAG_NAME) {
+                        echo "Detected release build triggered from tag ${env.TAG_NAME}."
+                        def isMajorMinorPatch = sh(
+                            script: "mchbuild -s version=${env.TAG_NAME} -g isMajorMinorPatch build.checkGivenSemanticVersion",
+                            returnStdout: true
+                        )
+                        if (isMajorMinorPatch != 'true') {
+                            currentBuild.result = 'ABORTED'
+                            error('Build aborted because release builds are only triggered for tags of the form <major>.<minor>.<patch>.')
                         }
+                        Globals.version = env.TAG_NAME
+                    } else
+                    {
+                        echo "Detected development build triggered from branch."
+                        Globals.version= sh(
+                            script: 'mchbuild -g semanticVersion build.getSemanticVersion',
```

Comment:
Did this work? I needed to [set the Git credentials correctly in order for this to work](https://github.com/C2SM-RCM/flexpart/blob/a43cdd26b0821f66a1480d8e505977b1f0f2ba81/Jenkinsfile#L85).

Reply:
The build went through ...

Reply:
The calculated semantic version looks weird though `1.0.12`